karapace/schema_reader.py

```diff
@@ -182,6 +184,12 @@ def get_schema_id(self, new_schema: TypedSchema) -> int:
             self.global_schema_id += 1
             return self.global_schema_id
 
+    def get_schema_id_if_exists(self, *, subject: Subject, schema: TypedSchema) -> Optional[int]:
+        return self._hash_to_schema_id_on_subject.get(hash((subject, schema.__str__())), None)
+
+    def _delete_from_schema_id_on_subject(self, *, subject: Subject, schema: TypedSchema) -> None:
+        self._hash_to_schema_id_on_subject.pop(hash((subject, schema.schema_str)), None)
```

Comment:
Get is using `TypedSchema`'s `__str__()`, but deleting using `schema.schema_str`. Why is that? Shouldn't this always use normalized string representation?

Reply:
I decided not to add documentation as now all subject+schema->id mapping functions are close together.